src/components/CabinList/index.js

```diff
@@ -1,21 +1,32 @@
-import React from 'react';
+import React, { useState } from 'react';
 import cabinData from '../../data/cabin';
 
-function CabinList() {
+function CabinList({ name }) {
   const click = (data) => {
-    console.log(data);
+    setSelectedCabinName(data);
   };
-  const cabinNameList1 = cabinData.map((data) => data.cabin);
-  const cabinNames = [...new Set(cabinNameList1)];
-  const cabin = cabinNames.map((data, index) => (
+  const [SelectedCabinName, setSelectedCabinName] = useState('');
```

Comment:
oh one more - soz.  SelectedCabin needs to be lowercase, e.g. selectedCabinName, it's a regular old variable.